File: leetgo/go/Offer-64/solution.go
// Created by k at 2023/06/20 10:23
// https://leetcode.cn/problems/qiu-12n-lcof/

package main

import (
	"bufio"
	"fmt"
	"os"

	. "github.com/j178/leetgo/testutils/go"
)

// @lc code=begin

func sumNums(n int) (ans int) {
	var recur func(n int) int
	recur = func(n int) int {
		_ = n > 1 && recur(n-1) > 0
		ans += n
		return ans
	}
	recur(n)

	return ans
}

// @lc code=end

func main() {
	stdin := bufio.NewReader(os.Stdin)
	n := Deserialize[int](ReadLine(stdin))
	ans := sumNums(n)

	fmt.Println("\noutput:", Serialize(ans))
}

File: leetgo/cpp/Offer-64/solution.cpp
// Created by k at 2023/06/20 10:17
// https://leetcode.cn/problems/qiu-12n-lcof/

#include <bits/stdc++.h>
#include "LC_IO.h"
using namespace std;

// @lc code=begin

class Solution {
public:
    int sumNums(int n) {
		char A[n][n+1];
		return sizeof(A)>>1;
    }
};
//ans=1+2+3+...+n
//   =(1+n)*n/2
//   =sizeof(bool a[n][n+1])/2
//   =sizeof(a)>>1


// @lc code=end

int main() {
	ios_base::sync_with_stdio(false);
	stringstream out_stream;

	int n; LeetCodeIO::scan(cin, n);

	Solution *obj = new Solution();

	auto res = obj->sumNums(n);

	LeetCodeIO::print(out_stream, res);
	cout << "
output: " << out_stream.rdbuf() << endl;

	delete obj;
	return 0;
}
